integration-tests/utils/lyle-utils.js

```diff
@@ -19,13 +19,27 @@ const fetchLyle = async (path, email) => {
   const url = `${lyleUrl}/api/${path}`
   const authClient = await makeAuthClient()
 
-  const res = await fetch(url, {
-    method: 'POST',
-    headers: { 'Content-Type': 'application/json', ...(await authClient.getRequestHeaders(url)) },
-    body: JSON.stringify({ email })
-  })
-
-  return res.json()
+  try {
+    const res = await fetch(url, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json', ...(await authClient.getRequestHeaders(url)) },
+      body: JSON.stringify({ email })
+    })
+    console.log(`fetchLyle: POST ${res.status} ${url}`)
+    if (res.ok) {
+      // response.status >= 200 && response.status < 300
+      return res.json()
+    }
+    // delegate non-2xx response to enclosing try/catch
+    const error = new Error()
+    Object.assign(error, { response: res })
+    throw error
```

Comment:
How would you feel about the following?

```suggestion
    if (res.ok) {
      return res.json()
    } else {
      // delegate non-2xx response to enclosing try/catch
      throw _.set('response', res, new Error())
    }
```

It would be a bit more concise; wouldn't mutate an object, and uses Lodash/FP for consistency. The first comment also seemed unnecessary.

Reply:
Like this?
```
if (res.ok) {
  return res.json()
}
// delegate non-2xx response to enclosing try/catch
throw _.set('response', res, new Error())
```	
Code looks clean. Just to confirm, do you know for sure if lodash/fp will work? We saw an issue with `_.forEach` in `jest-reporter.js` so I don't know if fp is a better choice.

Reply:
Yes, we've been happily using Lodash/FP throughout the codebase. I suspect the root cause for the issue you mentioned was something else.

Reply:
Updated.